### PR TITLE
Refactor redundancy_zone/upgrade_version out of client meta

### PR DESCRIPTION
**Pull request by kyhavlov**
This removes the `RedundancyZoneTag`/`UpgradeVersionTag` fields and adds corresponding autopilot and server config values to store the values in Serf tags instead of client metadata.

### CODE REVIEW

File: api/operator_autopilot.go [hunk 1]
```diff
@@ -54,6 +54,45 @@ type AutopilotConfiguration struct {
 	ModifyIndex uint64
 }
 
+func (u *AutopilotConfiguration) MarshalJSON() ([]byte, error) {
+	type Alias AutopilotConfiguration
+	return json.Marshal(&struct {
```
Comment (dadgar) [inline api/operator_autopilot.go]: Why is this needed? `time.Duration` should serialize fine?

File: command/agent/config-test-fixtures/basic.hcl [hunk 2]
```diff
@@ -83,6 +83,8 @@ server {
 	retry_interval = "15s"
 	rejoin_after_leave = true
 	non_voting_server = true
+    redundancy_zone = "foo"
```
Comment (dadgar) [inline command/agent/config-test-fixtures/basic.hcl]: formatting

File: command/operator_autopilot_set.go [hunk 3]
```diff
@@ -21,9 +19,9 @@ func (c *OperatorAutopilotSetCommand) AutocompleteFlags() complete.Flags {
 			"-max-trailing-logs":         complete.PredictAnything,
 			"-last-contact-threshold":    complete.PredictAnything,
 			"-server-stabilization-time": complete.PredictAnything,
-			"-redundancy-zone-tag":       complete.PredictAnything,
+			"-enable-redundancy-zones":   complete.PredictAnything,
```
Comment (dadgar) [inline command/operator_autopilot_set.go]: These should be `PredictNothing` since they are boolean options

File: nomad/autopilot.go [hunk 4]
```diff
@@ -10,13 +10,40 @@ import (
 	"github.com/hashicorp/serf/serf"
 )
 
+const (
+	AutopilotRZTag      = "ap_zone"
```
Comment (dadgar) [inline nomad/autopilot.go]: Throw some comments on these

File: website/source/guides/cluster/autopilot.html.md [hunk 5]
```diff
@@ -193,11 +199,11 @@ to voters and demoting the old servers. After this is finished, the old servers
 safely removed from the cluster.
 
 To check the Nomad version of the servers, either the [autopilot health]
-(/api/operator.html#read-health) endpoint or the `Nomad members`
+(/api/operator.html#read-health) endpoint or the `nomad members`
 command can be used:
 
 ```
-$ Nomad members
+$ nomad members
```
Comment (dadgar) [inline website/source/guides/cluster/autopilot.html.md]: These examples should be changed to be `nomad server-members`. The output is slightly different

**Comment by dadgar** [comment]
Check out the test failures. There are build failures

File: api/operator_autopilot.go (hunk 1 above)
Comment (kyhavlov) [reply api/operator_autopilot.go]: By default it serializes to the underlying int64 (which is in nanoseconds) instead of the "3h" or "200ms" format

File: website/source/guides/cluster/autopilot.html.md [hunk 6]
```diff
@@ -193,11 +199,11 @@ to voters and demoting the old servers. After this is finished, the old servers
...
 
 To check the Nomad version of the servers, either the [autopilot health]
-(/api/operator.html#read-health) endpoint or the `Nomad members`
+(/api/operator.html#read-health) endpoint or the `nomad members`
 command can be used:
 
 ```
-$ Nomad members
+$ nomad server-members
 Node   Address         Status  Type    Build  Protocol  DC
```
Comment (dadgar) [inline website/source/guides/cluster/autopilot.html.md]: There is no type on server-members:
```
Name                 Address    Port  Status  Leader  Protocol  Build      Datacenter  Region
alex-2.local.global  127.0.0.1  4648  alive   true    2         0.8.0-dev  dc1         global
```
